src/transformers/configuration_utils.py

```
@@ -1058,8 +1058,8 @@ def dict_dtype_to_str(self, d: dict[str, Any]) -> None:
         if d.get("dtype") is not None:
             if isinstance(d["dtype"], dict):
                 d["dtype"] = {k: str(v).split(".")[-1] for k, v in d["dtype"].items()}
-            elif not isinstance(d["dtype"], str):
-                d["dtype"] = str(d["dtype"]).split(".")[1]
+            elif not isinstance(d["dtype"], str) and (str_dtype := str(d["dtype"])).count("."):
+                d["dtype"] = str_dtype.split(".")[1]
```

ig the "dtype" in Emu3 is a token id, so maybe we can check that it's not an int and we don't ignore errors if "dtype" was set as `float32`